pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -568,7 +568,9 @@ func (asw *actualStateOfWorld) SetVolumeGloballyMounted(
 
 	volumeObj.globallyMounted = globallyMounted
 	volumeObj.deviceMountPath = deviceMountPath
-	volumeObj.devicePath = devicePath
+	if devicePath != "" {
+		volumeObj.devicePath = devicePath
+	}
```

Comment:
Do we really need this? If we prevent concurrent operation on same volume - won't a pod that uses same volume will pretty much cause another `MountDevice` operation?

Reply:
Yes, we need it. Consider UnmountDevice is in progress and a new pod arrives.

1. UnmountDevice finishes. This calls `MarkDeviceAsUnmounted` which clears the devicePath, i.e. calls `SetVolumeGloballyMounted(..., devicePath="", ...)`
2. New MountDevice (WaitForAttach) starts and sees empty devicePath -> problems.

Reply:
sorry to put a comment this late. very good catch of this bug. I think to avoid such bugs, it is better to not reuse SetVolumeGloballyMounted for both MarkDeviceAsMounted and MarkDeviceAsUnmounted (this function just to set the bool globallyMounted to false so no need to pass devicePath or deviceMountPath)